### PR TITLE
add op "ZerosLike" and "Any"support

### PR DESCRIPTION
**Pull request by zhijxu-MS**
add op "ZerosLike" and "Any"support

### CODE REVIEW

File: tf2onnx/tfonnx.py [hunk 1]
```diff
@@ -1664,21 +1664,35 @@ def floormod_op(ctx, node, name, args):
...
         reduce_dim = [reduce_dim[0]]
+    for i in reduce_dim:
+        utils.make_sure(i >= 0, "negative reduce axis is not supported in onnx for now")
 
-    utils.make_sure(all(i >= 0 for i in reduce_dim), "negative axis is not supported in onnx")
     cast = ctx.make_node(op_type="Cast", inputs=[node.input[0]], attr={"to": onnx_pb.TensorProto.INT32})
     keepdims = helper.get_attribute_value(node.get_attr("keep_dims"))
-    reduce_min = ctx.make_node(op_type="ReduceMin", inputs=cast.output, attr={"axes": reduce_dim, "keepdims": keepdims})
-    res = ctx.make_node(op_type="Cast", inputs=reduce_min.output, attr={"to": onnx_pb.TensorProto.BOOL},
+    op_type = "ReduceMin" if node.type == "All" else "ReduceSum"
```
Comment (pengwa) [inline tf2onnx/tfonnx.py]: you can put ReduceMin in the mapping 

_OPSET_6 = { 
    "All": (reduce_logic_op, ["ReduceMin"]),

File: tf2onnx/tfonnx.py [hunk 2]
```diff
@@ -1664,21 +1664,35 @@ def floormod_op(ctx, node, name, args):
...
-def all_op(ctx, node, name, args):
+def reduce_logic_op(ctx, node, name, args):
     # T output = All(T x, list(int) reduce_indices, @bool keepdims)
+    # T output = Any(T x, list(int) reduce_indices, @bool keepdims)
     reduce_dim = node.inputs[1].get_tensor_value()
     if isinstance(reduce_dim, np.ndarray):
         reduce_dim = reduce_dim.tolist()
     else:
         reduce_dim = [reduce_dim[0]]
+    for i in reduce_dim:
```
Comment (pengwa) [inline tf2onnx/tfonnx.py]: use all() for a check

or you can use node.input[0]'s shape length to calculate the negative axis's corresponding positive axis number.

File: tf2onnx/tfonnx.py [hunk 3]
```diff
@@ -1664,21 +1664,35 @@ def floormod_op(ctx, node, name, args):
...
+    op_type = "ReduceMin" if node.type == "All" else "ReduceSum"
+    reduce_node = ctx.make_node(op_type=op_type, inputs=cast.output, attr={"axes": reduce_dim, "keepdims": keepdims})
+    res = ctx.make_node(op_type="Cast", inputs=reduce_node.output, attr={"to": onnx_pb.TensorProto.BOOL},
                         name=node.name, outputs=node.output)
-    return [cast, reduce_min, res]
+    return [cast, reduce_node, res]
+
+
+def zeroslike_op(ctx, node, name, args):
+    # T output = zeros_like(T x)
```
Comment (pengwa) [inline tf2onnx/tfonnx.py]: ZerosLike(T x)

File: tf2onnx/tfonnx.py (hunk 1 above)
Comment (zhijxu-MS) [reply tf2onnx/tfonnx.py]: maybe ["All", "ReduceMin"], i knew the mechanism while i thought the op's conversion logic is not complex, so it's better to put it in function so reader not needs to figure out what is "args"

File: tf2onnx/tfonnx.py (hunk 2 above)
Comment (zhijxu-MS) [reply tf2onnx/tfonnx.py]: done

File: tf2onnx/tfonnx.py (hunk 1 above)
Comment (pengwa) [reply tf2onnx/tfonnx.py]: dont' think so. you can check Max/Min/Mean's implementation. 

and have you ever consider use existing reduce_op to do the implementation?

File: tf2onnx/tfonnx.py [hunk 4]
```diff
@@ -1803,8 +1802,8 @@ def zeroslike_op(ctx, node, name, args):
 
 _OPSET_6 = {
     "AddN": (direct_op, ["Sum"]),
-    "All": (reduce_logic_op, []),
-    "Any": (reduce_logic_op, []),
+    "All": (reduce_logic_op, ["All", "ReduceMin"]),
```
Comment (pengwa) [inline tf2onnx/tfonnx.py]: I don't mean this. you can check how reduce_op is reused by Sum/Prod/Min/Mean/Max. Basically it kept original node.

**Pull request merged**